app/services/api/v3/ecf_partnerships_query.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module Api
+  module V3
+    class ECFPartnershipsQuery
+      attr_reader :lead_provider, :params
+
+      def initialize(lead_provider:, params:)
+        @lead_provider = lead_provider
+        @params = params
+      end
+
+      def partnerships
+        scope = lead_provider.partnerships.includes(:school, :cohort, :delivery_partner)
+        scope = scope.where("partnerships.cohort_id IN (?)", with_cohorts.map(&:id)) if filter[:cohort].present?
```

Comment:
I'd prefer avoiding hardcoding the table's name in the query and let ActiveRecord so its thing as much as possible.
```ruby
scope.where(partnerships: { cohort:  with_cohorts} ) if filter[:cohort].present?
scope.where(partnerships: { delivery_partner: [delivery_partner_id_filter]} if delivery_partner_id_filter.present?
```